capsules/src/console.rs

```diff
@@ -204,15 +205,18 @@ impl<'a, U: UART> Client for Console<'a, U> {
         // Either print more from the AppSlice or send a callback to the
         // application.
         self.tx_buffer.replace(buffer);
-        self.in_progress.take().map(|appid| {
+        self.in_progress.get().map(|appid| {
+            self.in_progress.set(None);
```

Comment:
What problem would come up if the user forgets to set(None)? It seems like if that is going to be a problem, paying one word to avoid it everywhere is worth it.